src/Propel/Runtime/Connection/ConnectionWrapper.php

```diff
@@ -357,6 +358,8 @@ public function setAttribute($attribute, $value)
             default:
                 $this->connection->setAttribute($attribute, $value);
         }
+
+        return true;
```

Comment:
Previous return was `null`, why provide `true` here?

Reply:
The contract dictates it.

Reply:
The contract is still open to anything. Only the annotation is boolean, the code itself is changing here.

Something is wrong here, I think the interface annotation is mixed up with the one from PDO

Reply:
The contract is the specification of the interface
While it might code wise violate that due to missing typehints, that does not mean it is valid to do so.

In these cases we have to check how it is used. Also the PDO one dictates the behavior here.
It expects the bool result so the inheritance here must also do that.

Reply:
/**
	 * (PHP 5 &gt;= 5.1.0, PHP 7, PECL pdo &gt;= 0.1.0)<br/>
	 * Set an attribute
	 * @link https://php.net/manual/en/pdo.setattribute.php
	 * @param int $attribute
	 * @param mixed $value
	 * @return bool <b>TRUE</b> on success or <b>FALSE</b> on failure.
	 */
	public function setAttribute ($attribute, $value) {}

Reply:
It's weird because I'd have assumed the implicit `null` return was always similar to `false`, so making it true is odd. However, always returning false is clearly wrong.

But your fix is probably the best choice.